File: opencv_algorithm/chapter_01/cpp/read_img.cpp
// read a picture use cpp

#include <opencv2/opencv.hpp>
#include <iostream>
using namespace std;
using namespace cv;

int main()
{
    cout << "reading image..." << endl;
    int a = 10;
    Mat img = imread("test.png");
    cout << img.size() << endl;

    if (!img.data){
        return -1;
    }
    imshow("row picture", img);
    waitKey(0);

    cout << "That's all!" << endl;
    return 0;
}

File: opencv_algorithm/chapter_01/python/read_img.py
# read a picture use python

import cv2
import sys

if __name__ == "__main__":
    if len(sys.argv) > 1:
        # 输入图像
        image = cv2.imread(sys.argv[1], cv2.IMREAD_ANYCOLOR)
        # 显示图像
        cv2.imshow('image', image)
        cv2.waitKey(0)
        cv2.destroyAllWindows()
    else:
        print("Usage: python firstOpencv2.py imageFile")    